src/daria/analysis/contouranalysis.py

```diff
@@ -45,7 +45,12 @@ def contour_length(
 
     # Extract boolean mask covering pixels of interest.
     if img_roi.img.dtype == bool:
-        mask: np.ndarray = img_roi.img
+        if values_of_interest is None:
+            mask: np.ndarray = img_roi.img
+        else:
+            mask = np.zeros(img_roi.img.shape[:2], dtype=bool)
```

Comment:
why are you taking the .shape[:2] here, and not above (in the if-statement)? If they are supposed to yield the same sized array, why not use zeros_like?

Reply:
It has become a habit to not use zeros_like. Failed to often assuming a wrong shape when working with images, I guess. Here it is safe though. I changed it, thanks.